lib/spack/spack/cmd/bootstrap.py

```diff
@@ -96,6 +97,12 @@ def setup_parser(subparser):
         default=False,
         help="show the status of dependencies needed to develop Spack",
     )
+    status.add_argument(
+        "--check",
+        action="store_true",
+        default=False,
+        help="exit with an error code if any packages are reported as missing",
+    )
```

Comment:
I think we can simplify the PR and just add the exit status of 1, without the need for an additional argument.